src/sagemaker/jumpstart/model.py

```diff
@@ -429,16 +429,22 @@ def retrieve_example_payload(self) -> JumpStartSerializablePayload:
             sagemaker_session=self.sagemaker_session,
         )
 
-    def set_deployment_config(self, config_name: Optional[str]) -> None:
+    def set_deployment_config(self, config_name: str, instance_type: str) -> None:
```

Comment:
do we need validation on checking whether config_name + instance_type combination exists? Or customers can just choose whatever instance types they want?

Reply:
Good point, this instance_type override happens to the entire model, essentially equilvant to re-initialize the model with the instance_type. I will add a logic to check if the instance_type is supported in the config.

Reply:
Updated